firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/NotificationFeedbackTrigger.kt

```diff
@@ -0,0 +1,215 @@
+import com.google.firebase.ktx.Firebase
+import com.googletest.firebase.appdistribution.testapp.NotificationFeedbackTrigger.SCREENSHOT_FILE_NAME
+import com.googletest.firebase.appdistribution.testapp.NotificationFeedbackTrigger.takeScreenshot
+import java.io.IOException
+
+@SuppressLint("StaticFieldLeak") // Reference to Activity is set to null in onActivityDestroyed
+object NotificationFeedbackTrigger : Application.ActivityLifecycleCallbacks {
+    private const val TAG: String = "NotificationFeedbackTrigger"
+    private const val FEEBACK_NOTIFICATION_CHANNEL_ID = "InAppFeedbackNotification"
+    private const val FEEDBACK_NOTIFICATION_ID = 1
+    const val SCREENSHOT_FILE_NAME =
+        "com.googletest.firebase.appdistribution.testapp.screenshot.png"
+
+    private var isEnabled = false
+    private var hasRequestedPermission = false
+    private var currentActivity: Activity? = null
+    private var requestPermissionLauncher: ActivityResultLauncher<String>? = null
+
+    fun initialize(application: Application) {
+        // Create the NotificationChannel, but only on API 26+ because
+        // the NotificationChannel class is new and not in the support library
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = NotificationChannel(
+                FEEBACK_NOTIFICATION_CHANNEL_ID,
+                application.getString(R.string.feedback_notification_channel_name),
+                NotificationManager.IMPORTANCE_HIGH
+            )
+            channel.description =
+                application.getString(R.string.feedback_notification_channel_description)
+            val notificationManager = application.getSystemService(
+                NotificationManager::class.java
+            )
+            notificationManager.createNotificationChannel(channel)
+        }
+        application.registerActivityLifecycleCallbacks(this)
+    }
+
+    fun enable(currentActivity: Activity? = null) {
+        this.currentActivity = currentActivity
+        isEnabled = true
+        if (currentActivity != null) {
+            showNotification(currentActivity)
+        }
+    }
+
+    fun disable() {
+        isEnabled = false
+        val activity = currentActivity
+        currentActivity = null
+        if (activity != null) {
+            cancelNotification(activity)
+        }
+    }
+
+    private fun showNotification(activity: Activity) {
+        if (ContextCompat.checkSelfPermission(activity, POST_NOTIFICATIONS) == PERMISSION_GRANTED) {
+            val intent = Intent(activity, TakeScreenshotAndTriggerFeedbackActivity::class.java)
+            intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY)
+            val pendingIntent =
+                PendingIntent.getActivity(activity, 0, intent, PendingIntent.FLAG_IMMUTABLE)
+            val builder = NotificationCompat.Builder(activity, FEEBACK_NOTIFICATION_CHANNEL_ID)
+                .setSmallIcon(R.drawable.test_app_icon)
```

Comment:
Maybe leave a TODO here to pick a better icon? I believe this is an icon in the SDK resources (not the app resources) that is used for tests. Why that's available at runtime to the SDK is another thing we should fix at some point.

If you want to try one from the test app maybe `R.mipmap.ic_launcher`?

Reply:
Done.